roxie/ccd/ccdserver.cpp

```diff
@@ -27220,6 +27220,30 @@ IRoxieServerActivityFactory *createRoxieServerKeyedJoinActivityFactory(unsigned
 
 //=================================================================================
 
+class CRoxieServerSoapActivityBaseFactory : public CRoxieServerActivityFactory
+{ 
+    friend class CRoxieServerSoapActivityBase;
+protected:
+    bool useBlacklist = true;
+    unsigned blDelay = (unsigned) -1;
+    unsigned blRetries = (unsigned) -1;
+    unsigned blConnectTimeout = (unsigned) -1;
+    StringBuffer blError;
+
+public:
+    CRoxieServerSoapActivityBaseFactory(unsigned _id, unsigned _subgraphId, IQueryFactory &_queryFactory, HelperFactory *_helperFactory, ThorActivityKind _kind, IPropertyTree &_graphNode)
+        : CRoxieServerActivityFactory(_id, _subgraphId, _queryFactory, _helperFactory, _kind, _graphNode, soapStatistics)
+    {
+        if (_graphNode.getPropBool("hint[@name='noblacklist']/@value", false))
+            useBlacklist = false;
+        blDelay = _graphNode.getPropInt("hint[@name='blacklistdelay']/@value", (unsigned) -1);       
```

Comment:
I think these values will override the defaults - so -1 will be used as the delay.

Reply:
The code works, but in an unexpected way,,,